### DISCUSSION TITLE
Active Learning - Delete previous label submission on 3DSlicer

### DISCUSSION THREAD

**Discussion by nicholasbuenger** [Project-MONAI/MONAILabel#1536] [General]
Hi all,

thank you for the creation of this great project.

I have a technical question about Active Learning. 

Is it possible to delete a previous submitted label of a series and restart the labeling with the Scribbles tools? In the uploaded image, I have for example two series and I have labeled one already (that is why 50% is shown under Active Learning).
Typical use case: If somebody had labeled incorrectly and pressed "Submit Label", couldn't she/he restore the original labels and restart labeling the first series?

Please let me know if the question is clear.

Best,
Nicholas

![image](https://github.com/Project-MONAI/MONAILabel/assets/141348622/ee8f151b-f6f3-46fa-9fb4-a2be0809523e)

**Comment by diazandr3s** [comment]
Hi @niklu-igloo,

Glad to hear you find this framework useful.

Currently, MONAI Label does not have a label management mechanism. This means, that once the label is submitted, it can't be changed from the viewer.  

Submitted labels go to the `labels/final` folder. You could remove that label from the folder and re-annotate the image.

Hope that helps,

**Reply by nicholasbuenger** [reply]
Hi @diazandr3s,

thank you for your reply and the clear piece of information.

I have already tried to remove the label from `~/.cache/monailabel/dicom/d428d22fe8e85c96b930ae3d4bf173eb/labels/final` and nothing changes from the 3DSlicer side: it still shows that 50% of the labels have been submitted. The above-mentioned folder is completely empty at the moment.

Have I modified the wrong folder?

**Reply by diazandr3s** [reply]
Hi @niklu-igloo,

I can see you are using DICOM images. For that, you'll need to set up a DICOM Web server (https://github.com/Project-MONAI/MONAILabel/tree/main/plugins/ohif#installing-orthanc-dicomweb) so MONAI Label can connect to it.
If you still want to use a local folder, you'll need to convert the images to either NIfTI or NRRD format.

Hope this helps,

**Reply by nicholasbuenger** [reply]
Thank you for the hint. I will read the documentation in detail.
I have actually already set up a DICOM Web server from which I get the slices and I have already performed inference on them, which worked perfectly. The only problem is the active learning part. I guess I have to find the `labels/final` folder within the DICOM Web server and modify it from there, instead of changing it in the local machine on which MONAILabel is running. 
Would that be the right approach?

**Reply by diazandr3s** [reply]
If you are using DICOM Web, there is no such thing as `labels/final`. That only applies to the local folder using the NIfTI or NRRD file format.

[DICOM Web](https://www.dicomstandard.org/using/dicomweb) is a set of APIs for storing, retrieving, searching and deleting DICOM files. MONAI Label uses this [DICOM Web client](https://github.com/ImagingDataCommons/dicomweb-client) to communicate with the DICOM Web server.

Here is the dicom datastore as well:  https://github.com/Project-MONAI/MONAILabel/blob/main/monailabel/datastore/dicom.py#L45

To delete a segmentation, you can either use those APIs or, if using Orthanc, you can manually delete them using the Orthanc UI

**Reply by nicholasbuenger** [reply]
Thank you for the clear explanation. I have managed to delete through APIs the DICOM files that MONAILabel has stored into the DICOM Web server and the "Active Learning" bar has successfully been updated to 0%.

Thank you very much for solving this issue.

Best,
Nicholas
